src/compas_view2/app/controller.py

```diff
@@ -31,18 +32,29 @@ class Controller:
     def __init__(self, app):
         self.app = app
 
-    def interactive(func):
-        def wrapped(self):
-            def add(obj):
-                if obj:
-                    self.app.add(obj)
-                    self.app.view.update()
-
-            worker = Worker(func, self)
-            worker.signals.result.connect(add)
-            Worker.pool.start(worker)
-        return wrapped
-
+    def interactive(action="add"):
```

Comment:
An action argument is added to the `interactive` decorator, indicating adding a new object or editing a selected object